frappe/event_streaming/doctype/event_producer/event_producer.py

```diff
@@ -25,14 +26,25 @@ def before_insert(self):
 		self.create_custom_fields()
 
 	def validate(self):
+		self.validate_event_subscriber()
 		if frappe.flags.in_test:
 			for entry in self.producer_doctypes:
 				entry.status = 'Approved'
 
+	def validate_event_subscriber(self):
```

Comment:
Jut tried on a newly installed site.

This runs after before_insert where create_event_consumer fails and kicks the user out with Password Not Found exception (If keys aren't generated beforehand). Am I missing anything?

Reply:
Fixed